### PR TITLE
lexicon learning: filtered empty g2p-generated prons

### PR DESCRIPTION
**Pull request by xiaohui-zhang**

### CODE REVIEW

File: egs/tedlium/s5_r2/local/run_learn_lex.sh [hunk 1]
```diff
@@ -72,7 +72,7 @@ if [ $stage -le 0 ]; then
     $data/train_vocab.txt | sort > $data/oov_train.txt || exit 1;
   steps/dict/apply_g2p.sh --var-counts 4 $data/oov_train.txt \
     $g2p_mdl_dir exp/g2p/oov_lex_train || exit 1;
-  cat exp/g2p/oov_lex_train/lexicon.lex | cut -f1,3 | \
+  cat exp/g2p/oov_lex_train/lexicon.lex | awk '{if (NF>=3) print $0}' | cut -f1,3 | \
```
Comment (danpovey) [inline egs/tedlium/s5_r2/local/run_learn_lex.sh]: Can you update the script steps/dict/learn_lexicon.sh to check for empty prons and die if they exist with a meaningful error message?

Comment (xiaohui-zhang) [inline egs/tedlium/s5_r2/local/run_learn_lex.sh]: done

File: egs/wsj/s5/steps/dict/learn_lexicon.sh [hunk 2]
```diff
@@ -168,7 +168,8 @@ if [ $stage -le 0 ]; then
     # create an empty list of g2p generated prons, if it's not given.
     touch $dir/lexicon_g2p.txt
   else
-    cp $lexicon_g2p $dir/lexicon_g2p.txt 2>/dev/null
+    cat $lexicon_g2p | awk '{if (NF<2) {print "There is an empty pronunciation in lexicon_g2p.txt." \
+      > "/dev/stderr"; exit 1} print NF}' > $dir/lexicon_g2p.txt 2>/dev/null
```
Comment (danpovey) [inline egs/wsj/s5/steps/dict/learn_lexicon.sh]: This won't work as intended unless you do `|| exit 1` after this command.

Comment (xiaohui-zhang) [inline egs/wsj/s5/steps/dict/learn_lexicon.sh]: sorry. fixed

File: egs/wsj/s5/steps/dict/learn_lexicon.sh [hunk 3]
```diff
@@ -168,7 +168,8 @@ if [ $stage -le 0 ]; then
     # create an empty list of g2p generated prons, if it's not given.
     touch $dir/lexicon_g2p.txt
   else
-    cp $lexicon_g2p $dir/lexicon_g2p.txt 2>/dev/null
+    cat $lexicon_g2p | awk '{if (NF<2) {print "There is an empty pronunciation in lexicon_g2p.txt." \
+      > "/dev/stderr"; exit 1} print NF}' > $dir/lexicon_g2p.txt || exit 1;
```
Comment (danpovey) [inline egs/wsj/s5/steps/dict/learn_lexicon.sh]: Also, this 'print NF' is not right.  It should be 'print $0'.  I think maybe you should test this.

Comment (xiaohui-zhang) [inline egs/wsj/s5/steps/dict/learn_lexicon.sh]: my bad.... tested

**Comment by danpovey** [comment]
Merging.  Thanks!

**Pull request merged**